include/aklite.h

```diff
@@ -0,0 +1,178 @@
+  /**
+   * Return the sha256 OStree hash of the Target.
+   */
+  std::string Sha256Hash() const { return sha256_; }
+  /**
+   * Return the FoundriesFactory CI build number or in TUF, custom.version.
+   */
+  int Version() const { return version_; }
+
+  /**
+   * Return TUF custom data for a Target.
+   */
+  Json::Value Custom() const { return custom_; }
+
+ private:
+  std::string name_;
+  std::string sha256_;
+  int version_;
+  Json::Value custom_;
+};
+
+/**
+ * The response from an AkliteClient call to CheckIn()
+ */
+class CheckInResult {
+ public:
+  enum class Status {
+    Ok,        // check-in was good
+    OkCached,  // check-in failed, but locally cached meta-data is still valid
+    Failed,    // check-in failed and there's no valid local meta-data
+  };
+  CheckInResult(Status status) : status(status) {}
+  Status status;
+  std::vector<TufTarget> Targets() const { return targets_; }
+  TufTarget GetLatest() const { return targets_.back(); }
+
+ private:
+  friend class AkliteClient;
+  std::vector<TufTarget> targets_;
+};
+
+/**
+ * The response from an AkliteClient call to Download()
+ */
+class DownloadResult {
+ public:
+  enum class Status {
+    Ok,
+    DownloadFailed,
+    VerificationFailed,
+  };
+  DownloadResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * The response from an AkliteClient call to Install
+ */
+class InstallResult {
+ public:
+  enum class Status {
+    Ok,
+    NeedsCompletion,
+    Failed,
+  };
+  InstallResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
+  Status status;
+  std::string description;
+};
+
+/**
+ * AkliteClient provides an easy-to-use API for users wanting to customize
+ * the behavior of aktualiz-lite.
+ */
+class AkliteClient {
+ public:
+  /**
+   * Construct a client instance pulling in config files from the given
+   * locations. ex:
+   *
+   *   AkliteClient c(AkliteClient::CONFIG_DIRS)
+   *
+   * @param config_dirs The list of files/directories to parse sota toml from.
+   */
+  AkliteClient(const std::vector<boost::filesystem::path> &config_dirs);
+  /**
+   * Used for unit-testing purposes.
+   */
+  AkliteClient(std::shared_ptr<LiteClient> client) : client_(client) {}
+
+  /**
+   * Performs a "check-in" with the device-gateay. A check-in consists of:
+   *  1) ask device-gateway for a new root.json - normally a 404.
+   *  2) ask device-gateway for timestamp and snapshot metadata.
+   *  3) pull down a new targets.json if needed
+   */
+  CheckInResult CheckIn(std::string hwid = "") const;
+
+  /**
+   * Return the active aktualizr-lite configuration.
+   */
+  boost::property_tree::ptree GetConfig() const;
+
+  /**
+   * Report network and hwinfo to server. This only reports data if the
+   * information has changed locally.
+   */
+  void SendTelemetry() const;
```

Comment:
This is something that might be use-case/client specific, I mean each client might wanna define it's own set of telemetry data to send.

Reply:
our backend isn't capable of arbitrary telemetry?

Reply:
then what the point to expose this functionality, just send it internally whenever akliteclient decides to do so.

Reply:
I suppose we could put it in the CheckIn method.

Reply:
I've just force pushed an approach that consolidates SendTelemetry and ReportAkliteConfiguration into the CheckIn method:

 d2a5e23...5fd8825

I also stumbled upon a bug we have with reportHwInfo on intel:  https://github.com/foundriesio/aktualizr-lite/pull/115/commits/0232f9acf951ea939642f1f21f6a3337e6b0dd27